loader/loader.c

```diff
@@ -3046,7 +3102,22 @@ static VkResult read_data_files_in_search_paths(const struct loader_instance *in
     // Add the remaining paths to the list
     if (NULL != override_path) {
         strcpy(cur_path_ptr, override_path);
+        cur_path_ptr += strlen(override_path);
+        if (NULL != additional_env) {
+            *cur_path_ptr++ = PATH_SEPARATOR;
+
+            copy_data_file_info(additional_env, NULL, 0, &cur_path_ptr);
+
+            // Remove the last path separator
+            --cur_path_ptr;
+            *cur_path_ptr = '\0';
```

Comment:
This doesn't fill me with confidence. Does `copy_data_file_info` always add a path separator after the last path? Does decreasing the pointer then slapping '\0' not overwrite the last character?

Reply:
Doesn't hurt to add a check.